src/storage/src/table/streaming_table/state_table.rs

```diff
@@ -826,6 +886,105 @@ where
             }
         }
     }
+
+    /// This function scans kv pairs from the `shared_storage` and
+    /// memory(`mem_table`) with optional pk_bounds. If a record exist in both `shared_storage` and
+    /// `mem_table`, result `mem_table` is returned according to the operation(RowOp) on it.
+    #[try_stream(ok = (Vec<u8>, Cow<'a, Row>), error = StorageError)]
+    async fn into_stream_with_pk(self) {
```

Comment:
May refactor the current implementation to yield `(Cow<'a, Vec<u8>>, Cow<'a, Row>)`, so
- `into_stream` is simply `stream.map(|(_pk, row)| row)`
- `into_stream_with_pk` is `stream.map(|(pk, row)| (pk.trim_vnode(), row)`

Reply:
Good idea